torchao/quantization/GPTQ/README.md

```diff
@@ -0,0 +1,74 @@
+# GPTQ
+
+GPTQ is a quantization technique that improves accuracy for various forms of quantization as introduced in the paper: https://arxiv.org/abs/2210.17323
+
+In general GPTQ requires a model, a quantization technique and calibration data. GPTQ then optimizes the quantization parameters and quantized weights so they are more accurate accross the calibration data.
+
+## API
+
+The api for this technique is as follows:
+
+
+```python
+from torchao.quantization import MultiTensorInputRecorder, Int4WeightOnlyGPTQQuantizer
+
+model = get_model() # user provided function
+
+# first gather inputs
+input_recorder = MultiTensorInputRecorder()
+for i in range(calibration_limit):
+    args = get_next_input() # user provided function
+    input_recorder(*args) # compare to model(*args)
+     # note: can do input_recorder(*args, **kwargs) if needed
+
+# then perform GPTQ
+quantizer = Int4WeightOnlyGPTQQuantizer() # quantization parameters like group_size can be set here
```

Comment:
also maybe adding some example on how people can adapt this to their own quantization might be useful as well I think